source/common/http/conn_manager_impl.cc

```diff
@@ -551,8 +551,9 @@ void ConnectionManagerImpl::RdsRouteConfigUpdateRequester::requestRouteConfigUpd
     requestVhdsUpdate(host_header, thread_local_dispatcher, std::move(route_config_updated_cb));
     return;
   } else if (parent_.snapped_scoped_routes_config_ != nullptr) {
-    Router::ScopeKeyPtr scope_key =
-        parent_.snapped_scoped_routes_config_->computeScopeKey(*parent_.request_headers_);
+    const auto& dynamic_metadata = parent_.connection()->streamInfo().dynamicMetadata();
```

Comment:
It has bitten us more times than I would like that we have different stream info: in this case we have both connection and stream level stream info. Is there ever going to be a case where we want to actually pull metadata from an HTTP filter? I'm not super familiar with this code, but for example, couldn't there be a filter that runs before the on_demand filter which could set some metadata? Do we want to make this more clear in the docs and potentially also prepare the API for supporting stream level stream info vs. connection level?

Reply:
This actually may have just bitten me. I didn't realize that this `StreamInfo` inside the connection was specific to the connection and information set in the HTTP filters won't show up here. I actually need dynamic metadata set in an HTTP filter for my use-case.

Would it make sense to have a distinct `metadata_extract_type` for connections and streams?  In the meantime, I'll try and make some changes to the API to allow us to specify which exact metadata type we want to look up. 

I think I'm going to also beef up those docs around this to make it more clear. It wasn't obvious to me at all that this is the case.